compiler/rustc_next_trait_solver/src/delegate.rs

```diff
@@ -1,12 +1,11 @@
-use std::fmt::Debug;
 use std::ops::Deref;
 
 use rustc_type_ir::fold::TypeFoldable;
 use rustc_type_ir::solve::{Certainty, Goal, NoSolution, SolverMode};
 use rustc_type_ir::{self as ty, InferCtxtLike, Interner};
 
 pub trait SolverDelegate:
-    Deref<Target: InferCtxtLike<Interner = <Self as SolverDelegate>::Interner>> + Sized
+    Deref<Target: InferCtxtLike<Interner = <Self as SolverDelegate>::Interner> + Sized> + Sized
```

Comment:
what made the inner `Sized` bound necessary? Alternatively, should `InferCtxtLike` simply have a `Sized` super trait bound?

Reply:
Since we pass `&**delegate` to `trait_ref_is_knowable`, which has a `Infcx: InferCtxtLike` bound, which isn't relaxed `?Sized`.